extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1706,101 +1752,89 @@ public void run()
-                            {
-                              // If existing value is null use the offset returned by task
-                              // otherwise use the max (makes sure max offset is taken from replicas)
-                              return existingOffsetInMap == null
-                                     ? partitionOffsets.getValue()
-                                     : Math.max(partitionOffsets.getValue(), existingOffsetInMap);
-                            }
-                          });
-                        }
-                      }
-                      return null;
-                    }
-                  }
-                          )
-              );
-            }
-          }
-          // not using futureTimeoutInSeconds as its min value is 120 seconds
-          // and minimum emission period for this metric is 60 seconds
-          Futures.successfulAsList(futures).get(30, TimeUnit.SECONDS);
-
-          // for each partition, seek to end to get the highest offset
-          // check the offsetsResponse map for the latest consumed offset
-          // if partition info not present in offsetsResponse then use lastCurrentOffsets map
-          // if not present there as well, fail the compute
-
-          long lag = 0;
-          for (PartitionInfo partitionInfo : partitionInfoList) {
-            long diff;
-            final TopicPartition topicPartition = new TopicPartition(ioConfig.getTopic(), partitionInfo.partition());
-            lagComputingConsumer.seekToEnd(ImmutableList.of(topicPartition));
-            if (offsetsResponse.get(topicPartition.partition()) != null) {
-              diff = lagComputingConsumer.position(topicPartition) - offsetsResponse.get(topicPartition.partition());
-              lastCurrentOffsets.put(topicPartition.partition(), offsetsResponse.get(topicPartition.partition()));
-            } else if (lastCurrentOffsets.get(topicPartition.partition()) != null) {
-              diff = lagComputingConsumer.position(topicPartition) - lastCurrentOffsets.get(topicPartition.partition());
-            } else {
-              throw new ISE("Could not find latest consumed offset for partition [%d]", topicPartition.partition());
-            }
-            lag += diff;
-            log.debug(
-                "Topic - [%s] Partition - [%d] : Partition lag [%,d], Total lag so far [%,d]",
-                topicPartition.topic(),
-                topicPartition.partition(),
-                diff,
-                lag
-            );
-          }
-          emitter.emit(
-              ServiceMetricEvent.builder().setDimension("dataSource", dataSource).build("ingest/kafka/lag", lag)
-          );
-        }
-        catch (InterruptedException e) {
-          // do nothing, probably we are shutting down
-        }
-        catch (Exception e) {
-          log.warn(e, "Unable to compute Kafka lag");
-        }
+    synchronized (lagComputingConsumerLock) {
```

Comment:
@gianm yeah that makes sense, I'll look into caching the results. 

And yeah, the Kafka request should be governed by the Kafka config `request.timeout.ms` which by default is 30s